test/Auction.test.js

```diff
@@ -163,6 +167,10 @@ describe("Auction", () => {
         await increaseTime(24000)
       })
 
+      it("should not self destruct", async () => {
+        expect(await isCodeAt(auction.address)).to.be.true
```

Comment:
We are testing `onOffer` view function and this function can not self-destruct. I'd move this check to "should be opened" test, just like we have in the previous context.